tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -35,13 +35,26 @@
 import xml.etree.ElementTree as ET
 import os
 import sys
+import re
```

Comment:
note: from all the refactorings you're doing, only the changes in this file `extract_metadata_from_bazel_xml.py` should be included in this PR. Everything else seems like cleanup/refactoring and it doesn't belong here.